Woodoku/tests/Entity/test_woodoku_board.py

```diff
@@ -0,0 +1,121 @@
+from typing import Tuple
```

Comment:
There are two approaches to balancing productivity and being thorough when testing. 
One is to test as much as possible and the other is to only test the core functionality.

I'd say that we should only test public methods and public classes, unless the private classes are complex enough to warrant a test. That means only testing Woodoku board and not the representation. But that's just a strategy to save time working on actually implementing features instead of always writing tests. 

Now, to address some issues with the tests that are already written. I'd say there need to be more cases. Empty, 1, 3, 5 blocks are just the basics, of course, you should include them, but the important tests are the edge cases. Like inserting/removing/checking invalid blocks. This should apply to all future test written that keep in mind some common errors

second note: perhaps the name of this file should be _test_WoodokuBoard.py_?

Reply:
We might need to reach an agreement before adding more tests to TestWoodokuRepresentation. Since the representation is a private data class, the private field will be accessed when testing each method isolated. 

There are three plans summarized according to the conversations above:

1. remove the current tests and test each method pairwisely with basics and more examples
2. keep the current tests and test more edge cases with the public getter of the private field `__board` in the representation class
3. leave the current tests as it is now and focus more on testing the public tests (ie: all other classes except `WoodokuBoardRepresentation`)

I am fine with any of the plans. You guys can vote.

Reply:
> 1. pairwisely

what does this mean?

Reply:
> 
> ```python
>     def test_add_blocks_add_one_block(self):
>         board = _WoodokuBoardRepresentation()
>         board.add_blocks(self.one_block_lst)
>         assert board.is_occupied(self.one_block_lst)
> ```
> 
> You might claim that in this above way, we are not testing `add_blocks` method alone. In fact, we do not need to. The only reason we call `add_blocks` is because we hope to get a `true` value when we call `is_occupied` later. Thus, as long as these two methods have consistent result, it should be acceptable.
> 
> Same logic applies to `remove_blocks` and `is_not_occupied`.


Similarly with the example above, I might test combination such as`add_block` + `is_occupied`/`is_not_occupied` and `remove_block` + `is_occupied/is_not_occupied`

Reply:
> Also, I found that the module name should be short, and all-lowercase, according to [doc](https://peps.python.org/pep-0008/#package-and-module-names). Then should we keep camelcase names in src/Entity or should we change the name of test files? I am fine with either way.

refered in new issue #30 referred. discussion should be moved there

Reply:
Well, I choose 1 + 2 + 3 hahah...

refactor the test written, add edge cases, and focus on public methods from now on, use your own judgement when using multiple non-tested methods in the same test function in balancing productivity and catching errors.